### PR TITLE
storage: address flakes in TestClientRetryNonTxn

### PR DESCRIPTION
**Pull request by eisenstatdavid**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/8549)

### CODE REVIEW

**Comment by tamird** [comment]
Your commit message should begin with the package name - that the PR includes the package name will not be persisted in the git log

**Comment by tamird** [comment]
Also, the verbose explanation you included in #8404 should be included in the *commit message* which addresses it. Finally, you need to include the text "Fixes #8404." in that same commit message so that Github will automatically close the issue when this PR is merged.

**Comment by eisenstatdavid** [comment]
Done

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8549#-:-KPEVFWc7XeKmsLv6A66:bb6un1j)*

File: internal/client/client_test.go [hunk 1]
```diff
@@ -267,6 +266,9 @@ func TestClientRetryNonTxn(t *testing.T) {
 					}
 					return errors.New("non-transactional client has not pushed txn yet")
 				})
+				if test.canPush {
```
Comment (tamird) [inline internal/client/client_test.go]: Does it make sense to add a comment here?

**Comment by eisenstatdavid** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[internal/client/client_test.go, line 269 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8549#-KPEPs4Rv0eTIOVQBB0d-r1-269:-KPEc5YViLOHY0ttEx7y:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/4371640fb2f0840a1bf615376e3d3bd644ebd575/internal/client/client_test.go#L269)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> Does it make sense to add a comment here?
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8549)*

**Comment by tamird** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8549#-:-KPEcyRsknzve28K6mYU:bnfp4nl)*

**Pull request merged**